doc/Coding_guidelines.md

```diff
@@ -613,7 +613,7 @@ Feel free to improve this document. Bring more examples and rules. But please tr
 
     - Always protect shared resource access if simultaneous read/write is possible with appropriate synchronization mechanism.
 
-    - Avoid time of check and time of usage mismatch. In multithread application value of shared variable may be changed at any moment by other thread.
+    - Avoid time of check and time of usage mismatch. In multithread application value of shared variable may be changed at any moment by other thread. If atomics are used, need to put their's modification inside Compare And Swap (CAS) loops with ```std::compare_exchange_weak``` (prefer this to ```std::compare_exchange_strong```).
```

Comment:
Didn't catch. Wheter we need looping or not depends on code logic. We need a loop when implementing something like `mutex.lock` but when we have simply `if (check) do this; else do that;` we don't. And recommendation about `std::compare_exchange_weak` over `std::compare_exchange_strong` is true only when we use loops.

Reply:
Well I actually mentioned that I'm talking about modification, not just usage. I meant something like atomic counter. Em, didn't get your last point, I can't imagine usage of any of the ```std::compare_exchange_X``` without loop